config/class-sitedetailsindex.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+namespace Automattic\VIP\Config;
+
+class SiteDetailsIndex {
+	private static $_instance;
+
+	private const LOG_FEATURE_NAME = 'site_details_index';
+
+	public static function instance() {
+		if ( ! ( static::$_instance instanceof SiteDetailsIndex ) ) {
+			static::$_instance = new SiteDetailsIndex();
+			static::$_instance->init();
+		}
+
+		return static::$_instance;
+	}
+
+	public function init() {
+		add_filter( 'vip_site_details_index_data', [ $this, 'set_env_and_core' ] );
+	}
+
+	/**
+	 * Given the site details from the vip_site_details_index_data filter, set the environment and core values
+	 *
+	 * @param {array} $site_details The current site details
+	 * @return {array} The new site details
+	 */
+	public function set_env_and_core( $site_details ) {
+		global $wp_version;
+
+		$site_id = 0;
+		if ( defined( 'VIP_GO_APP_ID' ) && VIP_GO_APP_ID ) {
+			$site_id = VIP_GO_APP_ID;
+		} else {
+			if ( defined( 'FILES_CLIENT_SITE_ID' ) && FILES_CLIENT_SITE_ID ) {
+				$site_id = FILES_CLIENT_SITE_ID;
+			}
+		}
+
+		// log2logstash has it's own timestamp field as well. However, if/when we ever decide
+		// to swap indexes, we'll need our own. Added here so that the migration will be easier
+		// and all the fields we need are included in the actual site details index data.
+		$site_details['timestamp'] = time();
+		$site_details['client_site_id'] = $site_id;
+		$site_details['environment_id'] = $site_id;
+		$site_details['environment_name'] = strval( get_bloginfo( 'name' ) );
+		$site_details['plugins'] = $this->get_plugin_info();
+		$site_details['core']['wp_version'] = strval( $wp_version );
+		$site_details['core']['blog_id'] = get_current_blog_id();
+		$site_details['core']['is_multisite'] = is_multisite();
+		
+		return $site_details;
+	}
+
+	/**
+	 * Gather up all the information about all the plugins.
+	 *
+	 * Does not contain anything from vip-go-mu-plugins
+	 */
+	public function get_plugin_info() {
+		// Needed or get_plugins can't be found in some instances
+		require_once( ABSPATH . 'wp-admin/includes/plugin.php' );
+
+		$all_plugins = get_plugins();
+		$active_plugins = get_option( 'active_plugins' );
+
+		$plugin_info = array();
+
+		foreach ( $all_plugins as $key => $value ) {
+			$active = in_array( $key, $active_plugins, true );
+
+			$plugin_info[] = array(
+				'name' => $value['Name'],
+				'version' => $value['Version'],
+				'active' => $active,
```

Comment:
Aside: one other data point that would be interesting is whether the plugin was enabled via code or UI. I think there's a helper function we can call to check this?

Reply:
Sure, easy enough.